get_smb_test.go

```diff
@@ -0,0 +1,410 @@
+package getter
+
+import (
+	"context"
+	urlhelper "github.com/hashicorp/go-getter/helper/url"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestSmbGetter_impl(t *testing.T) {
+	var _ Getter = new(SmbGetter)
+}
+
+func TestSmbGetter_Get(t *testing.T) {
```

Comment:
I'm not seeing it in here -- isn't it also allowed to access a smb share with just a // instead of a smb://?  should we test these paths?

Reply:
This is allowed for windows, I'm not sure if it's allowed for Linux users I think it's not. [Here](https://ubuntu.com/tutorials/install-and-configure-samba#4-setting-up-user-accounts-and-connecting-to-share) there's an Ubuntu tutorial showing how to access smb shared folder from Linux and Windows machines.
Anyways, this case is handled by the FileGetter already, the smb getter is only for when the url has `smb` scheme. 

If the URL is `//dir/file` there's no way to know the files is shared using smb or CIFS protocol, for example, so we assume this is a file path.

I know it works because I tested it locally on Windows, and besides that the smb-getter transforms `smb://host/dir/file` to `//host/dir/file` for windows users and uses the FileGetter to resolve this. There's no test for this because it has to run on Windows. Do you have any thoughts on how to test it?